src/complex/Utilities/OStreamUtilities.cpp

```diff
@@ -0,0 +1,968 @@
+#include "OStreamUtilities.hpp"
+#include "StringUtilities.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataObject.hpp"
+#include "complex/DataStructure/DataPath.hpp"
+#include "complex/DataStructure/DataStructure.hpp"
+#include "complex/DataStructure/IDataArray.hpp"
+#include "complex/DataStructure/INeighborList.hpp"
+#include "complex/DataStructure/NeighborList.hpp"
+#include "complex/DataStructure/StringArray.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
+
+#include <algorithm>
+#include <fstream>
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+using namespace complex;
+
+void OStreamUtilities::OutputFunctions::neighborListImplWrapper(DataStructure& dataStructure, PrintMatrix2D& matrixRef, DataObject::Type type, const DataPath& path, bool hasIndex, bool hasHeaders)
```

Comment:
```suggestion
void OStreamUtilities::OutputFunctions::neighborListImplWrapper(const DataStructure& dataStructure, PrintMatrix2D& matrixRef, DataObject::Type type, const DataPath& path, bool hasIndex, bool hasHeaders)
```
`LoadNeighborListToMatrixImpl` takes a `NeighborList` by const ref so `DataStructure` should be passed by const ref also.